problem_builder/tests/unit/test_instructor_tool.py

```diff
@@ -0,0 +1,226 @@
+            matching_blocks = [block for block in block_tree if block['id'] == usage_id]
+            self.assertEqual(len(matching_blocks), 1)
+            # - Is `name` of that entry set to value of current child's preferred_attr?
+            matching_block = matching_blocks.pop()
+            if not attrs:
+                self.assertEqual(matching_block['name'], usage_id)
+            else:
+                preferred_attr = child.get('preferred_attr')
+                self.assertEqual(matching_block['name'], attrs[preferred_attr])
+
+    def test_student_view_block_tree_choice_blocks_excluded(self):
+        """
+        Check if `student_view` method excludes 'pb-choice' blocks when
+        building block tree for render context.
+        """
+        # Pretend that all blocks in self.block.children are of type 'pb-choice:
+        self.block.runtime.id_reader = Mock()
+        self.block.runtime.id_reader.get_block_type.return_value = 'pb-choice'
+
+        instructor_tool.loader = Mock()
+        instructor_tool.loader.render_template.return_value = u''
+
+        self.block.student_view(context={})
+        block_tree = instructor_tool.loader.render_template.call_args[0][1]['block_tree']
+
+        # Check size of block_tree: Should only include root block
+        self.assertEqual(len(block_tree), 1)
+
+    def test_student_view_block_tree_eligible(self):
+        """
+        Check if `student_view` method correctly marks MCQ, Rating, and
+        Answer blocks as eligible when building tree for render
+        context.
+
+        A block is eligible if its type is one of {'pb-mcq', 'pb-rating', 'pb-answer'}.
+        """
+        # Pretend that all blocks in self.block.children are eligible:
+        self.block.runtime.id_reader = Mock()
+        self.block.runtime.id_reader.get_block_type.return_value = random.choice(['pb-mcq', 'pb-rating', 'pb-answer'])
+
+        instructor_tool.loader = Mock()
+        instructor_tool.loader.render_template.return_value = u''
+
+        self.block.student_view(context={})
+        block_tree = instructor_tool.loader.render_template.call_args[0][1]['block_tree']
+
+        # Check size of block_tree: All blocks should be included
+        num_blocks = len(self.block.children) + 1
+        self.assertEqual(len(block_tree), num_blocks)
+
+        # Check if all blocks are eligible:
+        self.assertTrue(all(block['eligible'] for block in block_tree))
+
+    def test_student_view_block_tree_ineligible(self):
+        """
+        Check if `student_view` method correctly marks blocks that aren't
+        MCQ, Rating, or Answer blocks as ineligible when building tree
+        for render context.
+        """
+        # Pretend that none of the blocks in self.block.children are eligible:
+        self.block.runtime.id_reader = Mock()
+        self.block.runtime.id_reader.get_block_type.return_value = 'pb-dummy-type'
```

Comment:
@itsjeyd nit: to be precise - it tests that it marks 'pb-dummy-type' block as ineligible. Might make sense to use parameterized test with a couple of more examples (2-3 would be enough I believe)